olive/cli/benchmark.py

```diff
@@ -0,0 +1,132 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# -----------------------------------------------------------------------------
+from argparse import ArgumentParser
+from copy import deepcopy
+
+from olive.cli.base import (
+    BaseOliveCLICommand,
+    add_input_model_options,
+    add_logging_options,
+    add_save_config_file_options,
+    add_shared_cache_options,
+    get_input_model_config,
+    update_shared_cache_options,
+)
+from olive.common.utils import set_nested_dict_value
+
+
+class BenchmarkCommand(BaseOliveCLICommand):
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser("benchmark", help="Evaluate the model using lm-eval.")
```

Comment:
Is this benchmark cli for lm-eval specifically? I mean, if in the future, we want to support other benchmark tools, but want to keep using this CLI. Then `lm-eval` should be an argument then. Like `--tools lm-eval`, But not a block for this PR for now as we can change this later.